src/main.cpp

```diff
@@ -5170,6 +5170,13 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             return true;
         }
 
+        if (mapBlockIndex.find(headers[0].hashPrevBlock) == mapBlockIndex.end()) {
+            // Doesn't connect, instead of DoSing in AcceptBlockHeader, request deeper headers
+            if (!IsInitialBlockDownload())
+                pfrom->PushMessage(NetMsgType::GETHEADERS, chainActive.GetLocator(pindexBestHeader), uint256());
```

Comment:
Yes, I just looked at the first example of "prev block not found" in the debug log of one my long-running listening nodes, and it's a peer that fails to send connecting headers on initial startup:

```
2016-05-30 12:50:07 Added connection to [redacted] peer=963241
2016-05-30 12:50:07 received: version (102 bytes) peer=963241
2016-05-30 12:50:07 send version message: version 70012, blocks=414057, us=0.0.0.0:8333, them=[redacted], peer=963241
2016-05-30 12:50:07 sending: version (102 bytes) peer=963241
2016-05-30 12:50:07 sending: verack (0 bytes) peer=963241
2016-05-30 12:50:07 receive version message: /[redacted subver I've never heard of before]/: version 70002, blocks=16863, us=[redacted], peer=963241, peeraddr=[redacted]
2016-05-30 12:50:07 sending: ping (8 bytes) peer=963241
2016-05-30 12:50:07 sending: addr (31 bytes) peer=963241
2016-05-30 12:50:07 initial getheaders (414056) to peer=963241 (startheight:16863)
2016-05-30 12:50:07 sending: getheaders (997 bytes) peer=963241
2016-05-30 12:50:07 sending: inv (37 bytes) peer=963241
2016-05-30 12:50:08 sending: inv (37 bytes) peer=963241
2016-05-30 12:50:08 received: verack (0 bytes) peer=963241
2016-05-30 12:50:08 received: getaddr (0 bytes) peer=963241
2016-05-30 12:50:08 received: ping (8 bytes) peer=963241
2016-05-30 12:50:08 sending: pong (8 bytes) peer=963241
2016-05-30 12:50:08 received: pong (8 bytes) peer=963241
2016-05-30 12:50:08 received: headers (162003 bytes) peer=963241
2016-05-30 12:50:08 ERROR: AcceptBlockHeader: prev block not found
2016-05-30 12:50:08 ProcessMessages(headers, 162003 bytes) FAILED peer=963241
```

Given that no new information was exchanged in the less than 2 seconds this interaction took, I think this would result in an infinite loop where my peer would send me ~160kb of data every 2 seconds...